packages/tupaia-web/src/features/Map/MarkerLayer/MarkerLayer.tsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import React from 'react';
+import { useParams } from 'react-router';
+import camelCase from 'camelcase';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { MarkerLayer as UIMarkerLayer } from '@tupaia/ui-map-components';
+import {
+  useEntitiesWithLocation,
+  useEntity,
+  useMapOverlayReport,
+  useMapOverlays,
+  useProject,
+} from '../../../api/queries';
+import { EntityCode } from '../../../types';
+import { processMeasureData } from './processMeasureData';
+import { useDefaultMapOverlay } from './useDefaultMapOverlay';
+
+const useNavigateToDashboard = () => {
+  const { projectCode } = useParams();
+  const location = useLocation();
+  const navigate = useNavigate();
+  const { data: project } = useProject(projectCode);
+
+  return (entityCode: EntityCode) => {
+    const link = {
+      ...location,
+      pathname: `/${projectCode}/${entityCode}/${project?.dashboardGroupName}`,
+    };
+    navigate(link);
+  };
+};
+
+const useEntitiesByMeasureLevel = (measureLevel?: string) => {
+  const { projectCode, entityCode } = useParams();
+  const getSnakeCase = (measureLevel?: string) => {
+    return measureLevel
+      ?.split(/\.?(?=[A-Z])/)
+      .join('_')
+      .toLowerCase();
+  };
+
+  return useEntitiesWithLocation(
+    projectCode,
+    entityCode,
+    {
+      params: {
+        includeRoot: false,
+        filter: {
+          type: getSnakeCase(measureLevel),
+          generational_distance: {
+            comparator: '<=',
+            comparisonValue: 2,
+          },
+        },
+      },
+    },
+    { enabled: !!measureLevel },
+  );
+};
+
+export const MarkerLayer = () => {
+  // set the map default overlay if there isn't one selected
+  useDefaultMapOverlay();
```

Comment:
This effect could go anywhere and I'm not really sure where it should go. I put it here so that it's close to where the report data is used.

Reply:
I agree it could go anywhere. I almost feel like it should be built into the links, but then it's possible a link could be navigated to in some way without a default overlay code being in it (e.g directly entering the url). Is `useDefaultMapOverlay` used anywhere else? Could it possibly be contained in the map component?